stdlib/public/core/AnyHashable.swift

```diff
@@ -193,7 +227,34 @@ extension AnyHashable : Equatable {
   ///   - lhs: A type-erased hashable value.
   ///   - rhs: Another type-erased hashable value.
   public static func == (lhs: AnyHashable, rhs: AnyHashable) -> Bool {
-    return lhs._box._isEqual(to: rhs._box)
+    // If they're equal, we're done.
+    if let result = lhs._box._isEqual(to: rhs._box) { return result }
+
+    #if _runtime(_ObjC)
+    // If one used a custom representation but the other did not, bridge
+    // the one that did *not* use the custom representation to Objective-C:
+    // if the bridged result has a custom representation, compare those custom
+    // custom representations.
+    if lhs._usedCustomRepresentation != rhs._usedCustomRepresentation {
+      // If the lhs used a custom representation, try comparing against the
+      // custom representation of the bridged rhs (if there is one).
+      if lhs._usedCustomRepresentation {
+        if let customRHS = _getBridgedCustomAnyHashable(rhs._box._base) {
+          return lhs._box._isEqual(to: customRHS._box) ?? false
+        }
+        return false
+      }
+
+      // Otherwise, try comparing the rhs against the custom representation of
+      // the bridged lhs (if there is one).
+      if let customLHS = _getBridgedCustomAnyHashable(lhs._box._base) {
```

Comment:
I didn't do that because I was being obsessive about keeping the "lhs" and "rhs" ordered. If you feel strongly that it doesn't matter, I could simplify this.